DocumentDataAPI/DocumentDataAPI/Data/Mappers/EntityMapExtensions.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Linq.Expressions;
+using System.Reflection;
+using Dapper.FluentMap.Mapping;
+
+namespace DocumentDataAPI.Data.Mappers;
+
+public static class EntityMapExtensions
+{
+    public static string GetMappedColumnName<T>(this EntityMap<T> entityMap, string propertyName) where T : class
```

Comment:
Så vidt jeg forstår, så giver man den selv, altså EntityMap in this case, som parameter og derved extender typen med denne nye metode? Er det nogenlunde forstået korrekt?

Reply:
Ikke helt. Det her tilføjer den her `GetMappedColumnName` metode til `EntityMap` klassen, så du f.eks. kan kalde den ved `new DocumentMap().GetMappedColumnName(nameof(DocumentModel.Date))`. Det returnerer så "date", fordi `DocumentModel.Date` er sat op til at blive mappet til `date`.

Reply:
Og årsagen til at tilføje det er for at have ét mindre sted hvor vi skal vedligeholde navne på kolonnerne i vores tabeller (fordi vi bare kan genbruge dem fra vores mappers)

Reply:
Det er mega smart, så vi kan sørge for at det er så uafhængigt af navnene i tabellerne som muligt. Men jeg forstod det som, at fordi man giver den sig selv med `this`, typen `EntityMap`, så extender man `EntityMap` typen med den nye metode `GetMappedColumnName`. Og fordi `DocumentMap` så arver fra `EntityMap`, så kan man kalde metoden derpå, som om, at den allerede eksisterede på `DocumentMap`.

Er min monke forståelse okay?

Reply:
Yes det er nøjagtig det der sker. Det er super smart!

Reply:
Det lyder smart! Så er der mindre ændringer der skal laves hvis columns i vores schemas ændrer navn. Men hvorfor bruger vi så ikke det her alle steder? Altså for eksempel i queries i repositories?

Reply:
Well det kan vi godt. Så vil jeg dog nok prøve at gøre det lidt mere "accessible", for det bliver meget hurtigt meget convoluted ellers. Imagine:
`$"insert into document_contents ({new DocumentContentMap().GetMappedColumnName(nameof(DocumentContent.DocumentId))}, {new DocumentContentMap().GetMappedColumnName(nameof(DocumentContent.Index))}, ...`

Reply:
Ellers kan man selvfølgelig gemme det som get-only properties i vores repositories. Det er fandme en god idé sarmi, det kommer med!

Reply:
Ja det bliver convoluted, men som du siger så kan vi gemme det et andet sted!

Reply:
god diskussion gutter

Reply:
og god tilføjelse hr. dam